modules/nf-core/gsea/gsea/main.nf

```diff
@@ -0,0 +1,80 @@
+    tuple val(meta), path("*/gsea_report_for_target_*.html")                                                                                , emit: report_htmls_target
+    tuple val(meta), path("*/ranked_gene_list*.tsv")                                                                                        , emit: ranked_gene_list
+    tuple val(meta), path("*/gene_set_sizes.tsv")                                                                                           , emit: gene_set_sizes
+    tuple val(meta), path("*/butterfly_plot.png")                                                                                           , emit: butterfly_plot
+    tuple val(meta), path("*/global_es_histogram.png")                                                                                      , emit: histogram
+    tuple val(meta), path("*/heat_map_1.png")                                                                                               , emit: heatmap
+    tuple val(meta), path("*/pvalues_vs_nes_plot.png")                                                                                      , emit: pvalues_vs_nes_plot
+    tuple val(meta), path("*/ranked_list_corr_2.png")                                                                                       , emit: ranked_list_corr
+    tuple val(meta), path("*/[!gene_set_size|gsea_report|ranked_gene_list]*.tsv")                                                           , emit: gene_set_tsv, optional: true
+    tuple val(meta), path("*/[!gsea_report|heat_map_corr_plot|index|pos_snapshot|neg_snapshot]*.html")                                      , emit: gene_set_html, optional: true
+    tuple val(meta), path("*/[!butterfly|enplot|global_es_histogram|gset_rnd_es_dist|heat_map|pvalues_vs_nes_plot|ranked_list_corr]*.png")  , emit: gene_set_heatmap, optional: true
+    tuple val(meta), path("*/*_snapshot*.html")                                                                                             , emit: snapshot, optional: true
+    tuple val(meta), path("*/enplot*.png")                                                                                                  , emit: gene_set_enplot, optional: true
+    tuple val(meta), path("*/gset_rnd_es_dist*.png")                                                                                        , emit: gene_set_dist, optional: true
+    tuple val(meta), path("*/*.zip")                                                                                                        , emit: archive, optional: true
+    path "versions.yml"                                                                                                                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def VERSION = '4.3.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def args = task.ext.args ?: [ ]
```

Comment:
Since you are referencing keys in `args`, I guess the default should be an empty map? That's quite unusual, though, it's almost always a simple string.

```suggestion
    def args = task.ext.args ?: [:]
```

Reply:
I'm not sure how the usual string method would have been useful in this context, I'd have had to parse the values out of the string (since the options were not passed through verbatim to gsea as is often the case), which the map helped me to avoid. 


But in any case I've now switched to the channel usage instead of the args map